### PR TITLE
autogenerates dh2048, private/public keypair if necessary

### PR DESCRIPTION
**Pull request by krageon**
Does so from ansible. I don't agree the service wrapper should do this - so now it does not. Generating the fingerprint and signature should also be trivial.

As for actually registering the service, I've not decided. It makes sense that the service does this on it's own if it needs to, so I'll probably leave things there.

### CODE REVIEW

File: zerooo-server.yml [hunk 1]
```diff
@@ -7,8 +7,13 @@
   vars:
     name: "zer.ooo"
     servicedirectory: "/usr/lib/{{ name }}/"
+    configdirectory: "{{ servicedirectory }}openvpn/"
+    privatekey: "{{ configdirectory }}server.key"
+    publickey: "{{ configdirectory }}server.pub"
+    diffiehellman: "{{ configdirectory }}dh2048.pem"
+    fingerprint: "{{ configdirectory }}fingerprint"
+    signature: "{{ configdirectory }}signature"
```
Comment (EaterOfCode) [inline zerooo-server.yml]: Where are these used?

Comment (krageon) [inline zerooo-server.yml]: Fingerpring/signature isn't yet. I'm still debating

**Comment by krageon** [comment]
It's now used to create fingerprint/signature files. No real reason to generate those more than once. We need to speak about making a config file - I don't think all of these need to be stored separately, and there is at least one more dynamic setting.

**Comment by EaterOfCode** [comment]
the fingerprint is indeed used multiple times but the signature is only used when registering: https://github.com/zerooo-vpn/zer.ooo-server/blob/master/main.sh#L115 are we sure we want to have that in a seperate file

**Comment by krageon** [comment]
It's arcane and weird to generate (so I'd rather not see the generation anywhere in the wrapper).

I suppose if I'm going to make this ansible script register the node and store the resulting communication public key in a file, I can generate the signature only when registering is still necessary - then the file would exist only temporarily or it would get generated as a variable prerequisite to registering.

**Comment by EaterOfCode** [comment]
It's not that arcane nor weird to generate, only because it's bash it looks like so, 

```
openssl rsautl -inkey "${PRIVATE_KEY}" -sign
```

is the obvious signing openssl command,

```
hexdump -ve '1/1 "%.2x"'
```

changes all the stdin to hex.

this means that the code in php is

```php
$fingerprint = "~ magic ~";
openssl_sign($fingerprint, $signature, "~ private key ~");
echo bin2hex($signature);
```

You could possibly add a notify to the creation of the private key and register there

**Pull request merged**